iquip/apps/monitor.py

```diff
@@ -1011,6 +1113,31 @@ def _setDDSSwitch(self, device: str, channel: int, on: bool):
         self.ddsSwitchThread.finished.connect(self.ddsSwitchThread.deleteLater)
         self.ddsSwitchThread.start()
 
+    @pyqtSlot(dict)
+    def _updateTTLStatus(self, status: Dict[str, Any]):
+        """Updates the TTL status.
+        
+        Args:
+            See _TTLStatusThread signals section.
+        """
+        for device, output in status["outputs"].items():
+            name = self.ttlToName[device]
+            self.ttlControllerFrame.ttlWidgets[name].outputChanged.emit(output)
+        for device, level in status["levels"].items():
+            name = self.ttlToName[device]
+            self.ttlControllerFrame.ttlWidgets[name].levelChanged.emit(level)
+        override = status["overriding"]
+        if override is not None:
+            self.ttlControllerFrame.overrideChanged.emit(override)
+
+    def _startTTLStatusThread(self):
+        """Creates and starts a new _TTLStatusThread instance."""
+        devices = tuple(self.ttlToName)
+        self.ttlStatusThread = _TTLStatusThread(self.proxy_ip, self.proxy_port, devices)
+        self.ttlStatusThread.fetched.connect(self._updateTTLStatus)
```

Comment:
Maybe this should be connected via `QueuedConnection`, since it will be emitted in a non-GUI thread.